### PR TITLE
update rapids version for 24.10 release

### PR DESCRIPTION
**Pull request by nvliyuan**
This pr is to update the spark-rapids script version to 24.10.0 and update the readme doc

### CODE REVIEW

**Comment by nvliyuan** [comment]
@viadea please help review. CC @jayadeep-jayaraman @cjac

**Comment by cjac** [comment]
Oh hey, thanks for the ping.  I'll check it out.

**Review by cjac — CHANGES_REQUESTED** [review]
Let's get the docs updated to reflect the versions of Dataproc supported in Q4 2024

Our current supported versions follow:
2.2-debian12
2.1-debian11
2.0-debian10
2.2-ubuntu22
2.2-ubuntu20
2.0-ubuntu18
2.2-rocky9
2.1-rocky8
2.0-rocky8

And if all of these platforms are not supported by spark-rapids, then we'll need to merge in my changes to dask-rapids.

File: spark-rapids/README.md [hunk 1]
```diff
@@ -20,15 +20,15 @@ RAPIDS Accelerator For Apache Spark is supported on Dataproc 2.0+ (Spark 3.0)+.
...
+    *   NVIDIA Volta™ GPU architecture or better (V100, T4, A10/A100, L4 and later)
     *   Multi-node clusters with homogenous GPU configuration
 *   Software Requirements
-    *   NVIDIA GPU driver 440.33+
-    *   CUDA v11.5/v11.0/v10.2/v10.1
+    *   NVIDIA GPU driver R470+
+    *   CUDA v11.0+
     *   NCCL 2.11.4+
-    *   Ubuntu 18.04, Ubuntu 20.04 or Rocky Linux 7, Rocky Linux8, Debian 10, Debian 11
+    *   Ubuntu 20.04, Ubuntu 22.04, CentOS 7, or Rocky Linux 8, Debian 10, Debian 11
```
[the remark this answers is not in the document]
Comment (nvliyuan) [reply spark-rapids/README.md:31]: Thanks for pointing it out, we don't support centos7 now, update a link in the doc for the soft/hardware requirements.

File: spark-rapids/README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ export CUDA_VER=11.5
 
 gcloud dataproc clusters create $CLUSTER_NAME  \
     --region $REGION \
-    --image-version=2.0-ubuntu18 \
+    --image-version=2.1-ubuntu20 \
```
[the remark this answers is not in the document]
Comment (nvliyuan) [reply spark-rapids/README.md:66]: update to 2.2 image

File: spark-rapids/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvliyuan) [reply spark-rapids/README.md]: File related [pr](https://github.com/NVIDIA/spark-rapids/pull/11656) for the doc issue.

**Comment by nvliyuan** [comment]
Hi @cjac , can we merge this pr?

**Comment by nvliyuan** [comment]
Hi @cjac , any update?

**Comment by nvliyuan** [comment]
Hi @cjac , not sure that can we merge this pr now?

**Comment by cjac** [comment]
I haven't tested it thoroughly yet.  I've been caught up in refactoring shared code into templates.  I would like to generate this file from components rather than copy/pasting between scripts.

Can you let me know what you think of #1282 please?

**Comment by nvliyuan** [comment]
Let's merge the pr for now, it is just a version update, thx

**Comment by cjac** [comment]
let me try it in my environment...

**Comment by cjac** [comment]
running without cuda-version specified produces a request for //developer.download.nvidia.com/compute/cuda/12.1.1/local_installers/cuda_12.1.1_530.30.02_linux.run

**Comment by cjac** [comment]
incorrectly specifying rapids-runtime as `--metadata rapids-runtime="RAPIDS"` produces usable error message.

**Comment by cjac** [comment]
@nvliyuan - do you mind if I commit to your branch?

**Comment by nvliyuan** [comment]
please feel free to commit, thanks

**Comment by cjac** [comment]
a re-re-re-run took 4m35.536s to complete ; This looks good to me.  Let me run it through the automated tests.

It looks like it built the kernel more than once.

NG code in templates/spark-rapids/ caches builds to GCS after the first run completes so subsequent similar runs will have less work to do.

[1] https://github.com/LLC-Technologies-Collier/initialization-actions/tree/template-gpu-20241219/templates/spark-rapids

**Comment by cjac** [comment]
there's a known problem with our build system.  Un momento por favor.

**Comment by cjac** [comment]
failure on 2.2-rocky9 ; I'll spin that up in my env.

**Comment by cjac** [comment]
oof.  I forgot DKMS took so long to run.  I think the test is timing out for waiting on `dnf -y -q module install nvidia-driver:latest-dkms` maybe

**Comment by cjac** [comment]
`dnf -y -q install cuda-toolkit` is taking a long time, too

**Comment by cjac** [comment]
the run takes 14m9.444s on rocky9
```
# echo $?
0
```

**Comment by cjac** [comment]
I'm sorry, I seem to have done something to the commit history here.  The diffstat looks very wrong at this point.

```
$ git diff master | diffstat
 CONTRIBUTING.md                         |   20 ++---
 cloudbuild/Dockerfile                   |   22 +++++-
 cloudbuild/presubmit.sh                 |    1 
 cloudbuild/run-presubmit-on-k8s.sh      |   34 +++++++--
 dask/dask.sh                            |  213 ++++++++++++++++++++++++++++++++++++++++++++--------------
 dask/test_dask.py                       |   14 +++
 gpu/Dockerfile                          |   40 +++++++++++
 gpu/README.md                           |   28 +++----
 gpu/bazel.screenrc                      |   11 +++
 gpu/env.json.sample                     |    7 +
 gpu/install_gpu_driver.sh               |  652 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++---------------------------------------------------------
 gpu/manual-test-runner.sh               |   77 +++++++++++++++++++++
 gpu/run-bazel-tests.sh                  |   24 ++++++
 gpu/test_gpu.py                         |  167 ++++++++++++++++++++++++++++++++-------------
 h2o/sample-script.py                    |   11 ---
 horovod/horovod.sh                      |    4 -
 horovod/test_horovod.py                 |   13 ++-
 hue/README.md                           |  118 ++++++++++++++++++++++++++++++++
 hue/another-query.png                   |binary
 hue/create-hive-table.png               |binary
 hue/hue-ui.png                          |binary
 hue/simple-hiveql.png                   |binary
 integration_tests/dataproc_test_case.py |   18 +++-
 rapids/BUILD                            |    2 
 rapids/Dockerfile                       |   40 +++++++++++
 rapids/bazel.screenrc                   |   17 ++++
 rapids/env.json.sample                  |    7 +
 rapids/manual-test-runner.sh            |   77 +++++++++++++++++++++
 rapids/rapids.sh                        |  814 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----------------------------------------------------------
 rapids/run-bazel-tests.sh               |   23 ++++++
 rapids/test_rapids.py                   |  137 +++++++++++--------------------------
 rapids/verify_rapids_dask.py            |   19 -----
 rapids/verify_rapids_dask_yarn.py       |   19 +++++
 spark-rapids/README.md                  |   19 -----
 spark-rapids/spark-rapids.sh            |   14 ++-
 35 files changed, 1943 insertions(+), 719 deletions(-)

```

**Comment by cjac** [comment]
my apologies.  ambiguous use of 'master' here.  When diffed against master's commit, 169e98e424d87833e10b7761130f08bfde4b4815 I see what I expect.

**Comment by cjac** [comment]
many tests have passed.  still standing by for full green run.

**Comment by cjac** [comment]
okay, that looks good.

**Review by cjac — APPROVED** [review]
LGTM.

**Pull request merged**